src/components/Header/Header.tsx

```diff
@@ -0,0 +1,120 @@
+import { signOut, useSession } from "next-auth/react";
+
+import { useState, useEffect, useCallback } from "react";
+
+import { Button, Switch } from "~/components/ui";
+
+import { Sun, Moon } from "lucide-react";
+import { cn } from "~/lib/utils";
+
+import { setIsDay, useAppDispatch } from "~/rtk";
+
+import { useRouter } from "next/router";
+import { useTranslation } from "next-i18next";
+
+import Link from "next/link";
+
+import Login from "~/components/Login/Login";
+
+export const Header = () => {
+  const { data: session } = useSession();
+  const dispatch = useAppDispatch();
+  const router = useRouter();
+  const { locale: curLocale, locales, push, pathname } = router;
+  const { t } = useTranslation("common");
+  const [display, setDisplay] = useState(false);
+
+  if (router.pathname === "/" && display === false) {
+    setDisplay(true);
+  } else if (router.pathname !== "/" && display === true) {
+    setDisplay(false);
+  }
+
+  const [headerScroll, setHeaderScroll] = useState("-sm");
+
+  const handleThemeSwitch = () => {
+    const isDark = document.body.classList.contains("dark") as boolean;
+    dispatch(setIsDay(isDark));
+  };
+
+  const handleLangSwitch = () => {
+    const newLocale = locales!.find((locale) => locale !== curLocale);
+    push(pathname, undefined, { locale: newLocale });
+  };
+
+  const handleScroll = useCallback(() => {
+    if (window.scrollY > 8) return setHeaderScroll("-lg");
+    if (window.scrollY > 6) return setHeaderScroll("-md");
+    if (window.scrollY > 4) return setHeaderScroll("");
+    if (window.scrollY <= 4) return setHeaderScroll("-sm");
+  }, []);
+
+  useEffect(() => {
+    window.addEventListener("scroll", handleScroll);
+    return () => window.removeEventListener("scroll", handleScroll);
+  }, [handleScroll]);
+
+  return (
+    <header
+      className={cn(
+        `shadow${headerScroll}`,
```

Comment:
tailwind classnames are not working like this. you need to have full classname in the codebase to make tailwind work and compile classname and include it in the bundle